spec/jobs/whitehall_document_import_job_spec.rb

```diff
@@ -46,61 +49,90 @@
     WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
   end
 
-  context "when a GdsApi::BaseError exception is raised" do
-    let(:error) { GdsApi::BaseError.new }
+  context "when the import fails" do
+    let(:error_message) { "import failed error" }
     before do
-      allow(WhitehallImporter::Import).to receive(:call).and_raise(error)
+      allow(WhitehallImporter::Import).to receive(:call).and_raise(error_message)
     end
 
-    it "retries the job" do
-      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
+    it "unlocks the document in Whitehall" do
+      request = stub_whitehall_unlock_document(
+        whitehall_migration_document_import.whitehall_document_id,
+      )
 
-      expect(WhitehallDocumentImportJob).to have_been_enqueued
+      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
+      expect(request).to have_been_requested
     end
 
-    it "logs the error when retries have been exhausted" do
-      perform_enqueued_jobs do
-        WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
-      end
+    it "updates the document import state to 'import_failed'" do
+      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
 
       whitehall_migration_document_import.reload
-      expect(whitehall_migration_document_import.error_log).to eq(error.inspect)
+      expect(whitehall_migration_document_import).to be_import_failed
     end
+  end
 
-    it "updates the document import state to 'import_failed' if the import failed" do
-      perform_enqueued_jobs do
-        WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
-      end
+  context "when the sync fails" do
+    let(:error_message) { "sync failed error" }
+    before do
+      allow(WhitehallImporter::Import).to receive(:call)
```

Comment:
Do we need to do this? I notice we have a global mock.

Reply:
Good spot - I've removed it now.